onnxruntime/contrib_ops/cpu/qlinear_global_average_pool.cc

```diff
@@ -88,19 +90,26 @@ Status QLinearGlobalAveragePool::Compute(OpKernelContext* context) const {
 
   const float x_scale = *(tensor_x_scale->Data<float>());
   const float y_scale = *(tensor_y_scale->Data<float>());
-  auto dtype = X.GetElementType();
-  switch (dtype) {
-    case ONNX_NAMESPACE::TensorProto_DataType_UINT8:
-      return ComputeQLinearGlobalAvgPool(X.Data<uint8_t>(), x_scale, *(tensor_x_zero_point->Data<uint8_t>()),
-                                Y.MutableData<uint8_t>(), y_scale, *(tensor_y_zero_point->Data<uint8_t>()),
-                                N, C, image_size, channels_last_, tp);
-    default:
-      ORT_THROW("Unsupported 'dtype' value: ", dtype);
-  }
+  return ComputeQLinearGlobalAvgPool(X.Data<T8Bits>(), x_scale, *(tensor_x_zero_point->Data<T8Bits>()),
+                                     Y.MutableData<T8Bits>(), y_scale, *(tensor_y_zero_point->Data<T8Bits>()),
+                                     N, C, image_size, channels_last_, tp);
```

Comment:
do you need logic to throw when TensorProto_DataType_FLOAT?

Reply:
the schema definition of QLinearGlobalAveragePool can guard the input to be int8 and uint8